packages/graph/hash_graph/lib/graph/src/api/rest/link_type.rs

```diff
@@ -191,20 +202,29 @@ async fn update_link_type<P: StorePool + Send>(
     body: Json<UpdateLinkTypeRequest>,
     pool: Extension<Arc<P>>,
 ) -> Result<Json<PersistedOntologyIdentifier>, StatusCode> {
-    let Json(UpdateLinkTypeRequest { schema, account_id }) = body;
-
-    let mut store = pool.acquire().await.map_err(|report| {
-        tracing::error!(error=?report, "Could not acquire store");
-        StatusCode::INTERNAL_SERVER_ERROR
-    })?;
-
-    let link_type: LinkType = schema.try_into().into_report().map_err(|report| {
+    let Json(UpdateLinkTypeRequest {
+        schema,
+        type_to_update,
+        account_id,
+    }) = body;
+
+    let new_type_id = VersionedUri::new(
+        type_to_update.base_uri().clone(),
+        type_to_update.version() + 1,
```

Comment:
I'm not very happy that we force an increment of one. If I remember correctly, we decided to not do it as either (a) people could choose to not publish a version, so they have a few revisions locally and only publish the latest, or (b) some revisions are also hosted elsewhere (with for whatever reason has the same base URI).
I don't want to block merging on this as this is a scenario we don't need to cover right now, but it would be great if we have a task covering this.

Reply:
Some good thoughts, I think they're not issues for a number of reasons:
1. We've talked about "unpublished" versions as being a UI nicety, and as expressed I really don't think they should live as _**types**_ within the graph. We've floated other options such as them being stored as entities linked to the relevant blocks, etc. I bring this up because even though we haven't figured out the correct approach for draft types, etc., I think we can rule out storing them as native types. It would greatly complicate the table schemas, validation logic, querying implementations, etc. So I think your point (a) is a non-issue.
1. There are two elements to your statement and I think it's worth clarifying them
    1. Types can be hosted elsewhere - This is correct, but it's worth noting that things that are hosted elsewhere will not live in the same tables or interfaces as types owned by the graph. We will persist metadata records (and cache copies) of the types that live elsewhere, but that is it. They **must** be managed differently because.
       - We cannot serve them, their "$id" will be some other domain we do not own
       - We know nothing about their ownership, the person who owns their domain, or owns the type in their system, may not have a HASH account, and we certainly don't have write access
   1. _Some revisions_ - I apologise if I'm being pedantic but there's a subtle implication here that's incorrect in my mind. A type is uniquely determined by its base domain. We do not have a way to express a change of base domain, and I do not think we ever will, (because there's no way to validate that the owner of the base domain is the same as that who owns the new one, etc.). Thus, there will never be a situation where some revisions of a type live on `example.com` and then new revisions of the type live at a domain hosted by the graph. If a type is hosted by the graph, then its full life cycle will be at the Graph's domain. And we will be good BP citizens and ensure that updates are incremental, because it plays nicer with tooling that tries to find the next and previous version, etc.